File: internal/entity/shweet.go
package entity

import "time"

type Shweet struct {
	ID        string    `json:"id"`
	UserID    string    `json:"-"`
	Message   string    `json:"message"`
	User      *User     `json:"user,omitempty"`
	CreatedAt time.Time `json:"created_at"`
}

type ShweetDetails struct {
	Shweet
	LikeCount     int  `json:"like_count"`
	ReshweetCount int  `json:"reshweet_count"`
	CommentCount  int  `json:"comment_count"`
	Liked         bool `json:"liked"`
	ReShweeted    bool `json:"reshweeted"`
}

File: frontend/src/models/shweet.ts
import User from "./user";

export type Timeline = Shweet[];

export default interface Shweet {
  id: string;
  message: string;
  user: User;
  created_at: Date;
}

export interface ShweetDetails extends Shweet {
  like_count: number;
  reshweet_count: number;
  comment_count: number;
  liked: boolean;
  reshweeted: boolean;
}
